### PR TITLE
Dialectify more AmdTraceRay* intrinsic

### PR DESCRIPTION
**Pull request by LLJJDD**
Dialectify all AmdTraceRay* intrinsic functions in SpirvLowerRayTracing,
and retire processLibraryFunc().

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 44a4b9fd6c4030d802cec55dbd42364522ec9c3b
<details><summary>CTS tests (Failed: 0/211351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35147/70456 (49.9%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 35309/70456 (50.1%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39024/70454 (55.4%)</li>
<li>Failed: 0/70454 (0.0%)</li>
<li>Not Supported: 31430/70454 (44.6%)</li><li>Warnings: 0/70454 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39058/70441 (55.4%)</li>
<li>Failed: 0/70441 (0.0%)</li>
<li>Not Supported: 31383/70441 (44.6%)</li><li>Warnings: 0/70441 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/interface/lgc/GpurtDialect.td [hunk 1]
```diff
@@ -107,3 +109,183 @@ def GpurtGetTriangleCompressionModeOp : GpurtOp<"get.triangle.compression.mode",
...
   let summary = "return the traceRay triangle compression mode";
 }
+
+def GpurtGetFlattenedGroupThreadIdOp : GpurtOp<"get.flattened.group.thread.id", [Memory<[]>, WillReturn]> {
+  let arguments = (ins);
+  let results = (outs I32:$result);
+  let summary = "return the flattened group thread ID";
+}
+
+def GpurtSetHitAttributesOp : GpurtOp<"set.hit.attributes", [Memory<[(write)]>, WillReturn]> {
```
Comment (nhaehnle) [inline lgc/interface/lgc/GpurtDialect.td:119]: This and the other `set.foo` operations can be `Memory<[(write InaccessibleMem)]>`, which is slightly more precise.

File: lgc/interface/lgc/GpurtDialect.td [hunk 2]
```diff
@@ -107,3 +109,183 @@ def GpurtGetTriangleCompressionModeOp : GpurtOp<"get.triangle.compression.mode",
...
+
+    - instNodeAddrHi: Current instance node address hi bits.
+
+    - anyHitCallType: Indication of calling behavior on any hit shader.
+
+    - geometryIndex: Current geometry index.
+  }];
+}
+
+def GpurtGetHitAttributesOp : GpurtOp<"get.hit.attributes", [Memory<[(write)]>, WillReturn]> {
```
Comment (nhaehnle) [inline lgc/interface/lgc/GpurtDialect.td:144]: This one can be `Memory<[(write ArgMem), (read InaccessibleMem)]>`

File: lgc/interface/lgc/GpurtDialect.td [hunk 3]
```diff
@@ -107,3 +109,183 @@ def GpurtGetTriangleCompressionModeOp : GpurtOp<"get.triangle.compression.mode",
...
+  let description = [{
+    Compiler notification of hit triangle node pointer.
+
+    - bvhAddress: The BVH address.
+
+    - nodePointer: Node pointer of hit triangle.
+  }];
+}
+
+def GpurtGetParentIdOp : GpurtOp<"get.parent.id", [WillReturn]> {
```
Comment (nhaehnle) [inline lgc/interface/lgc/GpurtDialect.td:279]: Should be possible to make this one explicitly `Memory<[]>` because it doesn't touch mutable state, right?

**Review by nhaehnle — COMMENTED** [review]
Thanks. The dialect op definitions can be tightened slightly, the rest LGTM.

File: lgc/interface/lgc/GpurtDialect.td (hunk 3 above)
Comment (nhaehnle) [reply lgc/interface/lgc/GpurtDialect.td:279]: You're right, I missed that.

**Comment by amdvlk-admin** [comment]
### Test summary for commit f1e9785d63cef1afbb9da6683bf5f39853bd125e
<details><summary>CTS tests (Failed: 0/211351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35147/70456 (49.9%)</li>
<li>Failed: 0/70456 (0.0%)</li>
<li>Not Supported: 35309/70456 (50.1%)</li><li>Warnings: 0/70456 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39024/70454 (55.4%)</li>
<li>Failed: 0/70454 (0.0%)</li>
<li>Not Supported: 31430/70454 (44.6%)</li><li>Warnings: 0/70454 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39058/70441 (55.4%)</li>
<li>Failed: 0/70441 (0.0%)</li>
<li>Not Supported: 31383/70441 (44.6%)</li><li>Warnings: 0/70441 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thanks!

**Pull request merged**